core/src/banking_stage/consumer.rs

```diff
@@ -34,6 +34,21 @@ use {
 /// Consumer will create chunks of transactions from buffer with up to this size.
 pub const TARGET_NUM_TRANSACTIONS_PER_BATCH: usize = 64;
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct RetryableIndex {
+    pub index: usize,
+    pub immediately_retryable: bool,
```

Comment:
these wind up in an `Option` too, making this `bool` a hokey tri-state. prefer an enum?
```rust
enum RetryableIndex {
    Drop,
    Future { index: usize },
    Immediate { index: usize },
}
```

Reply:
ah, ofc first non-context line in the fold is the filter_map()... i suppose the Drop variant is mostly useless then. enum usually superior to bool for self-doc/readability and maintenance, but i guess pick your poison

Reply:
Ah yeah the `filter_map` there is on the txs so we only have `RetryableIndex`es for txs we actually attempted to process.

I'll stick with bool for now. using enum would be better if we expected there could be more options in the future. But it's either immediately retryable or not - not much in-between 😄